sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -981,6 +982,19 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     spec.map { case (k, v) => partCols.find(_.equalsIgnoreCase(k)).get -> v }
   }
 
+  private def isAlreadyExistsException(e: Throwable): Boolean = {
+    val maxDepth = 4
```

Comment:
did you test with different hive versions? one idea is to re-throw the exception in `HiveClientImpl`, and we can test it in `VersionsSuite`.

Reply:
done. Added a test to `VersionsSuite`.